lnwallet/rpcwallet/sign_coordinator.go

```diff
@@ -0,0 +1,913 @@
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) MuSig2RegisterNonces(_ context.Context,
+	in *signrpc.MuSig2RegisterNoncesRequest,
+	_ ...grpc.CallOption) (*signrpc.MuSig2RegisterNoncesResponse,
+	error) {
+
+	req := &walletrpc.SignCoordinatorRequest_MuSig2RegisterNoncesRequest{
+		MuSig2RegisterNoncesRequest: in,
+	}
+
+	return processRequest(
+		s, 0,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) (
+			*signrpc.MuSig2RegisterNoncesResponse, error) {
+
+			rpcResp := resp.GetMuSig2RegisterNoncesResponse()
+			if rpcResp == nil {
+				return nil, ErrUnexpectedResponse
+			}
+
+			return rpcResp, nil
+		},
+	)
+}
+
+// MuSig2Sign sends a MuSig2SignRequest to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) MuSig2Sign(_ context.Context,
+	in *signrpc.MuSig2SignRequest,
+	_ ...grpc.CallOption) (*signrpc.MuSig2SignResponse, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_MuSig2SignRequest{
+		MuSig2SignRequest: in,
+	}
+
+	return processRequest(
+		s, 0,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) (*signrpc.MuSig2SignResponse,
+			error) {
+
+			rpcResp := resp.GetMuSig2SignResponse()
+			if rpcResp == nil {
+				return nil, ErrUnexpectedResponse
+			}
+
+			return rpcResp, nil
+		},
+	)
+}
+
+// SignMessage sends a SignMessageReq to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignMessage(_ context.Context,
+	in *signrpc.SignMessageReq,
+	_ ...grpc.CallOption) (*signrpc.SignMessageResp, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignMessageReq{
+		SignMessageReq: in,
+	}
+
+	return processRequest(
+		s, 0,
+		func(reqId uint64) walletrpc.SignCoordinatorRequest {
+			return walletrpc.SignCoordinatorRequest{
+				RequestId:       reqId,
+				SignRequestType: req,
+			}
+		},
+		func(resp *signerResponse) (*signrpc.SignMessageResp, error) {
+			signResp := resp.GetSignMessageResp()
+			if signResp == nil {
+				return nil, ErrUnexpectedResponse
+			}
+
+			return signResp, nil
+		},
+	)
+}
+
+// SignPsbt sends a SignPsbtRequest to the remote signer and waits for the
+// corresponding response.
+//
+// NOTE: This is part of the RemoteSignerRequests interface.
+func (s *SignCoordinator) SignPsbt(_ context.Context,
+	in *walletrpc.SignPsbtRequest,
+	_ ...grpc.CallOption) (*walletrpc.SignPsbtResponse, error) {
+
+	req := &walletrpc.SignCoordinatorRequest_SignPsbtRequest{
+		SignPsbtRequest: in,
+	}
+
+	return processRequest(
+		s, 0,
```

Comment:
nit: we could define a constant `noTimeout = 0` to make this a bit more explicit.